shell.nix

```diff
@@ -0,0 +1,409 @@
+{ nixroot ? (import <nixpkgs> { }), defaultLv2Plugins ? false, lv2Plugins ? [ ]
+, releaseMode ? false, enableKeyfinder ? true, enableDjinterop ? false
+, buildType ? "auto", cFlags ? [ ], useClang ? false, useClazy ? false
+, buildFolder ? "build", qtDebug ? false }:
+let
+  inherit (nixroot)
+    pkgs lib makeWrapper clang-tools cmake fetchurl fetchgit glibcLocales
+    nix-gitignore python3 python37Packages;
+
+  llvm_packages = if builtins.hasAttr "llvmPackages_10" pkgs then
+    pkgs.llvmPackages_10
+  else
+    pkgs.llvmPackages;
+
+  clangVersion = builtins.map builtins.fromJSON
+    (lib.versions.splitVersion llvm_packages.clang.version);
+  glibcVersion = builtins.map builtins.fromJSON
+    (lib.versions.splitVersion pkgs.glibc.version);
+
+  # llvm < 10.0.2 and glibc >= 2.31 do not work with -fast-math
+  fastMathBug = (builtins.elemAt clangVersion 0) <= 10
+    && (builtins.elemAt clangVersion 1) <= 0 && (builtins.elemAt clangVersion 2)
+    < 2 && (builtins.elemAt glibcVersion 0) <= 2
+    && (builtins.elemAt glibcVersion 1) >= 31;
+
+  stdenv = if useClang then llvm_packages.stdenv else pkgs.stdenv;
+
+  qtDeps = if qtDebug then [
+    (pkgs.enableDebugging ((pkgs.qt5.override {
+      debug = true;
+      developerBuild = true;
+    }).qtbase))
+    pkgs.qt5.qtscript
+    pkgs.qt5.qtsvg
+    pkgs.qt5.qtx11extras
+    pkgs.qt5.qtdoc
+  ] else [
+    pkgs.qt5.qtbase
+    pkgs.qt5.qtdoc
+    pkgs.qt5.qtscript
+    pkgs.qt5.qtsvg
+    pkgs.qt5.qtx11extras
+  ];
+
+  cmakeBuildType = if buildType == "auto" then
+    (if releaseMode then "RelWithDebInfo" else "Debug")
+  else
+    buildType;
+
+  allCFlags = cFlags ++ [
+    ("-DKEYFINDER=" + (if enableKeyfinder then "ON" else "OFF"))
+    ("-DENGINEPRIME=" + (if enableDjinterop then "ON" else "OFF"))
+    ("-DCMAKE_BUILD_TYPE=" + cmakeBuildType)
+  ] ++ (if useClang && fastMathBug then [ "-DFAST_MATH=off" ] else [ ]);
+
+  git-clang-format = pkgs.stdenv.mkDerivation {
```

Comment:
why is that in here? Doesn't `pre-commit` work on nkxOS?

Reply:
when i started this patch, the pre-commit shipped with 19.xx was to old for our configuration and commit testing was not possible. I think I had problems backporting to a derivate so I installed it ugly in the shell. removed now.